### PR TITLE
Support PHP 7.2 - 8.0

### PR DESCRIPTION
**Pull request by dhirtzbruch**
The latest tagged release does not support PHP 7.3 - PHP 8.0 due to out-of-date dependencies.
This merge request aims to drop support of EOL PHP versions and instead add support for recent versions.

I assume that -if this MR should be accepted- a new major release should be tagged.

### CODE REVIEW

File: tests/PerformanceTest/IniReadingBench.php [hunk 1]
```diff
@@ -1,57 +1,59 @@
...
-    public function native()
+    public function benchNative()
     {
         parse_ini_string($this->ini, true);
     }
 
     /**
-     * @iterations 10000
+     * @BeforeMethods("setUp", "createIniString")
+     * @Iterations(5)
```
Comment (sgiehl) [inline tests/PerformanceTest/IniReadingBench.php:35]: Is there any reason for lowering the iterations count for those tests?

File: composer.json [hunk 2]
```diff
@@ -9,14 +9,14 @@
...
         }
     },
     "require": {
-        "php": ">=5.3.3"
+        "php": ">=7.2"
     },
     "require-dev": {
-        "phpunit/phpunit": "^4.8.36",
-        "athletic/athletic": "0.1.*"
+        "phpunit/phpunit": "^7.5.20",
```
Comment (sgiehl) [inline composer.json:19]: Can't we directly switch to PHPUnit 8? According to https://phpunit.de/supported-versions.html it should support PHP 7.2

File: tests/PerformanceTest/IniReadingBench.php (hunk 1 above)
Comment (dhirtzbruch) [reply tests/PerformanceTest/IniReadingBench.php:35]: No, sorry, I just had lowered that to speed up while switching to phpbench. Fixed in f753034c.

File: composer.json (hunk 2 above)
Comment (dhirtzbruch) [reply composer.json]: Of course, fixed in 3dec719449af36a87856d9c9ae1ef82f99b3f242.

File: phpunit.xml [hunk 3]
```diff
@@ -22,4 +18,8 @@
         </whitelist>
     </filter>
 
+    <logging>
+        <log type="coverage-clover" target="clover.xml"/>
+        <log type="coverage-html" target="coverage/"/>
+    </logging>
```
Comment (sgiehl) [inline phpunit.xml:24]: Guess we could actually remove that. We used to use coveralls for a while. But actually that isn't running anymore for a while already. We can consider adding a coverage service later again...

Comment (dhirtzbruch) [reply phpunit.xml:24]: Personally I like using the coverage-html for local development. I'd not need clover, of course.
However, we can of course remove both if you prefer.

Comment (sgiehl) [reply phpunit.xml:24]: Locally that's fine. It's simply useless to have travis generate those files at the moment.

Comment (dhirtzbruch) [reply phpunit.xml]: Of course, fixed in 2cbe2c8fed2d67fe64b588874bcdea7d126a1f7e

**Comment by tsteur** [comment]
As long as there are no regressions and php 7.2 is still supported should be all good

**Comment by diosmosis** [comment]
I have no exceptions.

**Pull request merged**